src/entity/projectile/WindCharge.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\entity\projectile;
+
+use pocketmine\block\Bell;
+use pocketmine\block\Block;
+use pocketmine\block\BlockTypeIds;
+use pocketmine\block\Button;
+use pocketmine\block\CakeWithCandle;
+use pocketmine\block\Candle;
+use pocketmine\block\ChorusFlower;
+use pocketmine\block\Door;
+use pocketmine\block\Lever;
+use pocketmine\block\Trapdoor;
+use pocketmine\entity\Entity;
+use pocketmine\event\entity\EntityDamageByChildEntityEvent;
+use pocketmine\event\entity\EntityDamageByEntityEvent;
+use pocketmine\event\entity\EntityDamageEvent;
+use pocketmine\event\entity\ProjectileHitEvent;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\RayTraceResult;
+use pocketmine\math\Vector3;
+use pocketmine\nbt\tag\CompoundTag;
+use pocketmine\network\mcpe\protocol\types\entity\EntityIds;
+use pocketmine\world\particle\WindExplosionParticle;
+use pocketmine\world\sound\DoorSound;
+use pocketmine\world\sound\FlintSteelSound;
+use pocketmine\world\sound\RedstonePowerOffSound;
+use pocketmine\world\sound\RedstonePowerOnSound;
+use pocketmine\world\sound\WindChargeBurstSound;
+
+use function ceil;
+use function floor;
+use function round;
+use function time;
+
+class WindCharge extends Throwable{
+	protected float $radius = 2.5;
+	protected float $damage = 0.5;
+
+	protected float $createdAt = 0;
+	protected float $lifetime = 300;
+
+	public static function getNetworkTypeId() : string{ return EntityIds::WIND_CHARGE_PROJECTILE; }
+
+	protected function getInitialDragMultiplier() : float{ return 0; }
+	protected function getInitialGravity() : float{ return 0; }
+
+	protected function initEntity(CompoundTag $nbt) : void{
+		parent::initEntity($nbt);
+
+		$this->createdAt = time();
```

Comment:
All minecraft related timing uses ticks, you should be doing like this

Reply:
Also there's already a variable that tracks entity's lived ticks: https://github.com/pmmp/PocketMine-MP/blob/585aee9386a787c95e73dd0a05ffca8329606b68/src/entity/Entity.php#L137